### PR TITLE
NIAD-1082: Date Time human readable

### PR DESCRIPTION
**Pull request by Broom94**
add human readable date format, refactor test data, add additional tests

### CODE REVIEW

File: service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/utils/DateFormatUtil.java [hunk 1]
```diff
@@ -23,54 +21,70 @@
...
     private static DateTimeFormatter getFormatStringForPrecision(TemporalPrecisionEnum precisionEnum) {
-        return HL7_FORMATS.getOrDefault(precisionEnum, HL7_SECONDS);
+        return HL7_FORMATS_COMPUTER_READABLE.getOrDefault(precisionEnum, HL7_SECONDS_COMPUTER_READABLE);
     }
 
     private static String convertWithPrecision(TemporalPrecisionEnum precisionEnum, Calendar calendar) {
         return calendar.toInstant().atZone(UK_ZONE_ID)
             .format(getFormatStringForPrecision(precisionEnum));
     }
+
```
Comment (z-uddin) [inline service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/utils/DateFormatUtil.java:78]: Was wondering if this whitespace could be trimmed down?

File: service/src/test/java/uk/nhs/adaptors/gp2gp/utils/DateFormatUtilTest.java [hunk 2]
```diff
@@ -53,6 +54,18 @@ public void When_FormattingDateTimeTypeToHl7_Expect_Hl7InUkZone(String input, St
...
+    @MethodSource("dateTextParams")
+    public void When_FormattingDateTimeTypeToText_Expect_Hl7InUkZone(String input, String expected) {
+        String observationJson = String.format(DATETIME_OBSERVATION_TEMPLATE, input);
+        Observation observation = FHIR_PARSER.parseResource(observationJson, Observation.class);
+
+        String actual = toTextFormat(observation.getValueDateTimeType());
+        assertThat(actual).isEqualTo(expected);
+    }
+
+
```
Comment (z-uddin) [inline service/src/test/java/uk/nhs/adaptors/gp2gp/utils/DateFormatUtilTest.java:67]: Was wondering if this whitespace could be trimmed down?

**Review by mmackiewicz1 — DISMISSED** [review]

**Comment by Broom94** [comment]
On hold failed dates for different timezone

File: service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/utils/DateFormatUtil.java [hunk 3]
```diff
@@ -1,76 +1,83 @@
...
 
 import org.hl7.fhir.dstu3.model.BaseDateTimeType;
 
-import com.google.common.collect.ImmutableMap;
-
-import ca.uhn.fhir.model.api.TemporalPrecisionEnum;
 import uk.nhs.adaptors.gp2gp.ehr.exception.EhrMapperException;
 
 public class DateFormatUtil {
+    public static final int NINE = 9;
```
Comment (mmackiewicz1) [inline service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/utils/DateFormatUtil.java:12]: What these constants represent? Maybe we could give them more specific names?

Comment (Broom94) [reply service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/utils/DateFormatUtil.java:12]: i represents the number nine

**Review by z-uddin — APPROVED** [review]

**Review by ElliotB-96 — APPROVED** [review]
Looks good

**Pull request merged**